engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
     * 2.) Raw inputs and it's coercion
     * 3.) Workflow Options
     * 4.) RuntimeAttributes w.r.t. a backend
+    *
     * @return The result of the validation as instance of `ValidationResult`
     */
-  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = {
-    (for {
-      namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      validatedInputs <- Future(validateInputs(workflowInputs, namespaceWithWorkflow))
-      validatedWorkflowOptions <- Future(validateWorkflowOptions(workflowOptions, CromwellBackend.backend()))
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
```

Comment:
@mcovarr My two cents: I think it's better to have a non-blocking calculation here, since this actor is now a singleton and might potentially receive simultaneous requests.